src/mono/sample/iOS/Program.csproj

```diff
@@ -4,140 +4,57 @@
     <OutputPath>bin</OutputPath>
     <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
     <TargetOS Condition="'$(TargetOS)' == ''">ios</TargetOS>
-    <TargetOS Condition="'$(TargetsiOSSimulator)' == 'true'">iossimulator</TargetOS>
-    <DeployAndRun Condition="'$(DeployAndRun)' == ''">true</DeployAndRun>
-    <iOSLikeDedup Condition="'$(TargetOS)' == 'ios'">true</iOSLikeDedup>
     <RuntimeIdentifier>$(TargetOS)-$(TargetArchitecture)</RuntimeIdentifier>
     <DefineConstants Condition="'$(ArchiveTests)' == 'true'">$(DefineConstants);CI_TEST</DefineConstants>
     <AppName>HelloiOS</AppName>
+    <MainLibraryFileName>Program.dll</MainLibraryFileName>
```

Comment:
If we set up `$(AssemblyName)` here, the `_CustomAppleAotCompileProps ` target at: https://github.com/dotnet/runtime/pull/86578/files#diff-560b34cc42222d3f6e5b53b0c70190ba4ab33181155b9157014b47c5ebb443aeR34 would become redundant and we could use:
```xml
<MainLibraryFileName>$(AssemblyName).dll</MainLibraryFileName>
```
instead

Reply:
Not sure if I followed you, but it sounds to me like two separate properties.

The `AssemblyName` property is used in AppleAppBuilder as the ProjectName. It is used across targets, and there is an issue if it is set globally. The `MainLibraryFileName` property is used in AppleAppBuilder as the main entry point and it may differ from the `AssemblyName`.

In case of the sample, the `AssemblyName` should be set to HelloiOS, while the `MainLibraryFileName` should be set to Program.dll.

Reply:
> Not sure if I followed you, but it sounds to me like two separate properties.

I thought we could do this globally:
```xml
<AssemblyName>$(AppName)</AssemblyName>
<MainLibraryFileName>$(AssemblyName).dll</MainLibraryFileName>
```
so it is simpler and does not require an extra Target, but you said there is an issue with that. 
What kind of issue were you getting?

Reply:
If set globally, Program.dll becomes HelloiOS.dll but the app should work as expected.

Reply:
Actually, based on the comment here: https://github.com/dotnet/runtime/blob/2855dd2d29c64df89498fbcb984653f596c1b976/src/tasks/AppleAppBuilder/AppleAppBuilder.cs#L35-L38
seems like: https://github.com/dotnet/runtime/blob/2855dd2d29c64df89498fbcb984653f596c1b976/src/mono/msbuild/apple/build/AppleBuild.targets#L258 is wrong and should be passing `$(AppName)` instead.

Although, I am not sure if that is going to break anything.

Reply:
I don't think `AppName` is used outside of the sample apps, and if it is not a predefined parameter in msbuild, we can do the following:
```
<AppName Condition="'$(AppName)' == ''">$(AssemblyName)</AppName>
```

Reply:
I would try changing `AppleBuild.targets` to pass `AppName` as the `ProjectName` and see if it breaks something.

Reply:
The tests are failing due to `The "AppleAppBuilderTask" task was not given a value for the required parameter "ProjectName"` as `AppName` is used only in the sample apps. Therefore, I can add 
```
<AppName Condition="'$(AppName)' == ''">$(AssemblyName)</AppName>
```